### PR TITLE
fix #560/broken ARM templates gives bad error message

### LINKED ISSUE #560: Invalid ARM template gets dropped from create request leading to poor error message
I was missing a comma in my ARM template and as a result, it seems that the CLI decided to send my create request onward to the REST API without it, passing only the parameters. This resulted in an error like the following:

*The deployment 'sfsamplesdeploy' must have either the TemplateLink or Template property set. Please see https://aka.ms/arm-deploy for usage details.*

When I tried the same operation in the old CLI, it told me exactly where the syntax error was in the template.

### PR DESCRIPTION
**Pull request by yugangw-msft**

### CODE REVIEW

File: src/azure/cli/tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        #test good case
+        with open(pathname, 'w') as good_file:
+            good_file.write('{"key1":"value1", "key2":"value2"}')
+        result = get_file_json(pathname)
+        self.assertEqual('value2', result['key2'])
+
+        #test error case
+        with open(pathname, 'w') as bad_file:
+            bad_file.write('{"key1":"value1" "key2":"value2"}')
+        self.assertRaises(Exception, get_file_json, pathname)
```
Comment (BurtBiel) [inline src/azure/cli/tests/test_util.py]: >self.assertRaises(Exception, get_file_json, pathname) [](start = 8, length = 53)

does it make sense to try and verify the error message here?

Comment (yugangw-msft) [inline src/azure/cli/tests/test_util.py]: Sure, i will check

**Comment by derekbekoe** [comment]
:shipit:

**Pull request merged**